tests/integ/test_cte.py

```diff
@@ -1005,6 +1006,25 @@ def test_time_series_aggregation_grouping(session, enable_sql_simplifier):
         session.sql_simplifier_enabled = original_sql_simplifier_enabled
 
 
+def test_table_select_cte(session):
+    table_name = random_name_for_temp_object(TempObjectType.TABLE)
+    df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+    df.write.save_as_table(table_name, table_type="temp")
+    df = session.table(table_name)
+    df_result = df.with_column("add_one", col("a") + 1).union(
+        df.with_column("add_two", col("a") + 2)
+    )
+    check_result(
+        session,
+        df_result,
+        expect_cte_optimized=False if session.sql_simplifier_enabled else True,
```

Comment:
Can we have test cases for when sql_simplifier_enabled is both True and False? The previous test uses a parameter for that.

Reply:
in our daily test, we actually runs all tests with sql simplifier disabled already here https://github.com/snowflakedb/snowpark-python/actions/runs/12169626312/job/33945760032, i don't think it is necessary to repeat the test here